File: Lab1/estadistica.py
import math

class Estadistica:
    def __init__(self, datos):
        self.__datos = datos

    def promedio(self):
        return sum(self.__datos) / len(self.__datos)

    def desviacion(self):
        prom = self.promedio()
        suma = sum((x - prom) ** 2 for x in self.__datos)
        return math.sqrt(suma / (len(self.__datos) - 1))

    def mostrar_resultados(self):
        print(f"El promedio es {self.promedio():.2f}")
        print(f"La desviación estándar es {self.desviacion():.5f}")

if __name__ == "__main__":
    numeros = list(map(float, input().split()))
    est = Estadistica(numeros)
    est.mostrar_resultados()

File: Lab1/Estadistica/src/prac6/Estadistica.java
package prac6;

import java.util.Scanner;

public class Estadistica {
    public static double promedio(double[] numeros) {
        double suma = 0;
        for (double num : numeros) {
            suma += num;
        }
        return suma / numeros.length;
    }

    public static double desviacion(double[] numeros) {
        double prom = promedio(numeros);
        double sumaDiferencias = 0;
        for (double num : numeros) {
            sumaDiferencias += Math.pow(num - prom, 2);
        }
        return Math.sqrt(sumaDiferencias / (numeros.length - 1));
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double[] numeros = new double[10];
        System.out.println("");
        for (int i = 0; i < 10; i++) {
            numeros[i] = scanner.nextDouble();
        }

        double prom = promedio(numeros);
        double desv = desviacion(numeros);
        
        System.out.printf("El promedio es %.2f\n", prom);
        System.out.printf("La desviación estándar es %.6f\n", desv);

        scanner.close();
    }
}